src/currency-field/CurrencyField.ts

```diff
@@ -0,0 +1,377 @@
+        }
+    }
+
+    _updateSymbolAndFormat(): void {
+        this._setSymbolAndSeparators();
+        this._setCurrencyFormat();
+    }
+
+    // Set the currency format that will be used to format the input value.
+    _setCurrencyFormat(): void {
+        try {
+            this._currencyFormat = new Intl.NumberFormat(this.locale, { currency: this.currency });
+        } catch (e) {
+            console.error('invalid locale or currency provided, falling back to default', e);
+            this._currencyFormat = new Intl.NumberFormat('en-US', { currency: 'USD' });
+        }
+    }
+
+    // Set the currency symbol and sets the currency format separator and cent separator depending on locale.
+    _setSymbolAndSeparators(): void {
+        /*
+         * Set the currency parts providing a default value of 1000.0 to return an array to get the currency separator and cents separator.
+         * Results can be the following based on the locale provided.
+         * ["$", "1", ",", "000", ".", "00"]
+         * ["1", ".", "000", ",", "00", " ", "$"]
+         */
+        let currencyPartsMap = [];
+        try {
+            currencyPartsMap = new Intl.NumberFormat(this.locale, {
+                style: 'currency',
+                currency: this.currency,
+                currencyDisplay: 'narrowSymbol'
+            })
+                .formatToParts(1000.0)
+                .map((c) => c.value);
+        } catch (e) {
+            console.error('invalid locale or currency provided, fallback will be used', e);
+            currencyPartsMap = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD', currencyDisplay: 'narrowSymbol' })
+                .formatToParts(1000.0)
+                .map((c) => c.value);
+        }
+        // Get the currency symbol assuming it is the value at the first index of the array.
+        const currencySymbol = currencyPartsMap[0];
+
+        // Check to see if value at first index of the array is a number Some locales will have the currency symbol at the last index.
+        if (parseInt(currencySymbol)) {
+            this._currencyFormatSeparator = currencyPartsMap[CURRENCY_SEPARATOR.AMOUNT];
+            this._currencyCentsSeparator = currencyPartsMap[CURRENCY_SEPARATOR.CENTS];
+            this._currencySymbol = currencyPartsMap[currencyPartsMap.length - 1];
+        } else {
+            this._currencyFormatSeparator = currencyPartsMap[CURRENCY_SEPARATOR.INVERSE_AMOUNT];
+            this._currencyCentsSeparator = currencyPartsMap[CURRENCY_SEPARATOR.INVERSE_CENTS];
+            this._currencySymbol = currencySymbol;
+        }
+    }
+
+    _parseAmount(value: string): number {
+        let cleanValue = '';
+
+        for (let i = 0; i < value.length; i++) {
+            const character = value.charAt(i);
+            if (/\d/.test(character)) {
+                cleanValue += character;
+            }
+        }
+
+        if (cleanValue) {
+            return parseInt(cleanValue);
```

Comment:
Should this not be parseFloat? Will post decimal values not be lost?

Reply:
As discussed part of the parseAmount function